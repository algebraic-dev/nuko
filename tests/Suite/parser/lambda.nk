let a: Int =
    \a => b
    \b => c
    \d => e
          f
          g