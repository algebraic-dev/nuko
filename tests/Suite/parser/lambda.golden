✓ That
 └Program
   ├typeDecls: Empty List
   ├letDecls:
   | └List
   |   └LetDecl
   |     ├declName: (Name: ValName a 0:4-0:5)
   |     ├declArgs: Empty List
   |     ├declBody:
   |     | └Block
   |     |   ├BlBind
   |     |   | ├Lam
   |     |   | | ├PId NoExt (Name: ValName a 1:5-1:6)
   |     |   | | ├Lower
   |     |   | | | ├Local:
   |     |   | | | | └Name: ValName b 1:10-1:11
   |     |   | | | └NoExt
   |     |   | | └1:4-1:11
   |     |   | └BlBind
   |     |   |   ├Lam
   |     |   |   | ├PId NoExt (Name: ValName b 2:5-2:6)
   |     |   |   | ├Lower
   |     |   |   | | ├Local:
   |     |   |   | | | └Name: ValName c 2:10-2:11
   |     |   |   | | └NoExt
   |     |   |   | └2:4-2:11
   |     |   |   └BlEnd
   |     |   |     └Lam
   |     |   |       ├PId NoExt (Name: ValName d 3:5-3:6)
   |     |   |       ├Block
   |     |   |       | ├BlBind
   |     |   |       | | ├Lower
   |     |   |       | | | ├Local:
   |     |   |       | | | | └Name: ValName e 3:10-3:11
   |     |   |       | | | └NoExt
   |     |   |       | | └BlBind
   |     |   |       | |   ├Lower
   |     |   |       | |   | ├Local:
   |     |   |       | |   | | └Name: ValName f 4:10-4:11
   |     |   |       | |   | └NoExt
   |     |   |       | |   └BlEnd
   |     |   |       | |     └Lower
   |     |   |       | |       ├Local:
   |     |   |       | |       | └Name: ValName g 5:10-5:11
   |     |   |       | |       └NoExt
   |     |   |       | └3:10-5:11
   |     |   |       └3:4-5:11
   |     |   └1:4-5:11
   |     ├declRet:
   |     | └TId
   |     |   ├Local:
   |     |   | └Name: TyName Int 0:7-0:10
   |     |   └NoExt
   |     └declExt: NoExt
   ├impDecls: Empty List
   └programExt: NoExt
